test/compiler.test.js

```diff
@@ -1014,6 +1014,176 @@ describe('compiler', function() {
       instructions = boot.compile(appdir.PATH);
       expect(instructions.config).to.not.have.property('modified');
     });
+
+    describe('for mixins', function() {
+      function verifyMixinsUsingOptions(sourceFile, mixinDirs) {
+        var appJS = appdir.writeFileSync(sourceFile, '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinDirs: mixinDirs
+        });
+
+        expect(instructions.mixins[0].sourceFile).to.eql(appJS);
+      }
+
+      it('supports `mixinDirs` option', function() {
+        verifyMixinsUsingOptions('mixins/other.js', ['mixins']);
+      });
+
+      it('resolves relative path in `mixinDirs` option', function() {
+        verifyMixinsUsingOptions('custom-mixins/vehicle.js',
+          ['./custom-mixins']);
+      });
+
+      it('resolves module relative path in `mixinDirs` option', function() {
+        verifyMixinsUsingOptions('node_modules/custom-mixins/vehicle.js',
+          ['custom-mixins']);
+      });
+
+      it('picks mixins from `mixinDirs` when provided', function() {
+        var OtherJS = appdir.writeFileSync('custom-mixins/other.js', '');
+        appdir.writeFileSync('mixins/default.js', '');
+
+        var instructions = boot.compile({
+          appRootDir: appdir.PATH,
+          mixinDirs: ['custom-mixins']
+        });
+
+        expect(instructions.mixins).to.have.length(1);
+        expect(instructions.mixins[0].sourceFile).to.eql(OtherJS);
+      });
+
+      describe('name normalization', function() {
+        beforeEach(function() {
+          appdir.writeFileSync('mixins/foo.js', '');
+          appdir.writeFileSync('mixins/time-stamps.js', '');
+          appdir.writeFileSync('mixins/camelCase.js', '');
+          appdir.writeFileSync('mixins/PascalCase.js', '');
+          appdir.writeFileSync('mixins/space name.js', '');
+        });
+
+        it('supports classify', function() {
+          var options = { appRootDir: appdir.PATH, normalization: 'classify',
+          mixinDirs: ['mixins'] };
+          var instructions = boot.compile(options);
+
+          var mixins = instructions.mixins;
+          var mixinNames = mixins.map(getNameProperty);
+
+          expect(mixinNames).to.eql(['CamelCase', 'Foo',
+            'PascalCase', 'SpaceName', 'TimeStamps']);
```

Comment:
Please use a different indentation, so that the list of items is not split in the middle. Few examples illustrating style that I find easier to read:

```js
expect(mixinNames).to.eql([
  'CamelCase', 'Foo', 'PascalCase', 'SpaceName', 'TimeStamps'
]);

expect(mixinNames)
  .to.eql(['CamelCase', 'Foo', 'PascalCase', 'SpaceName', 'TimeStamps']);
```

The same comment applies to all normalization tests below.